lib/active_merchant/billing/gateways/cyber_source_rest.rb

```diff
@@ -127,16 +146,40 @@ def build_credit_request(amount, payment, options)
         end.compact
       end
 
-      def add_code(post, options)
-        return unless options[:order_id].present?
+      def create_customer_request(payment, options)
+        { buyerInformation: {}, clientReferenceInformation: {}, merchantDefinedInformation: [] }.tap do |post|
+          post[:buyerInformation][:merchantCustomerId] = options[:merchant_customer_id] if options[:merchant_customer_id]
+          post[:buyerInformation][:email] = options[:email].presence || 'null@cybersource.com'
+          add_code(post, options)
+        end.compact
+      end
 
-        post[:clientReferenceInformation][:code] = options[:order_id]
+      def create_instrument_identifier_request(payment, options)
+        instrument_identifier = {
+          card: {
+            number: payment.number
+          }
+        }
+        commit('instrumentidentifiers', instrument_identifier)
       end
 
-      def add_customer_id(post, options)
-        return unless options[:customer_id].present?
+      def create_customer_payment_instrument_request(payment, options, customer_id, instrument_identifier)
+        post = {}
+        post[:default] = 'true'
+        post[:card] = {}
+        post[:card][:type] = CREDIT_CARD_CODES[payment.brand.to_sym]
+        post[:card][:expirationMonth] = payment.month.to_s
+        post[:card][:expirationYear] = payment.year.to_s
+        add_address(post, payment, options[:billing_address], options, :billTo, nil)
+        post[:instrumentIdentifier] = {}
+        post[:instrumentIdentifier][:id] = instrument_identifier.params['id']
+        commit("customers/#{customer_id}/payment-instruments", post)
+      end
 
-        post[:paymentInformation][:customer] = { customerId: options[:customer_id] }
+      def add_code(post, options)
```

Comment:
Would suggest this, since "add_code" sounds like it could mean anything 😄 
```suggestion
      def add_order_id(post, options)
```